File: data/python/tests/test_Schematics_model_import_data.py
import unittest
from function_Schematics_model_import_data import Product

class TestProductModel(unittest.TestCase):

    def test_valid_data(self):
        product = Product()
        data = '{"name": "Laptop", "price": 1000}'
        expected = "{'name': 'Laptop', 'price': 1000, 'category': 'General'}"
        self.assertEqual(product.import_and_validate(data), expected)

    def test_missing_required_field(self):
        product = Product()
        data = '{"price": 1000}'
        result = product.import_and_validate(data)
        self.assertIn("name", result)

    def test_default_category(self):
        product = Product()
        data = '{"name": "Tablet", "price": 500}'
        native = product.import_and_validate(data)
        self.assertIn("'category': 'General'", native)

    def test_all_fields_provided(self):
        product = Product()
        data = '{"name": "Headphones", "price": 200, "category": "Electronics"}'
        expected = "{'name': 'Headphones', 'price': 200, 'category': 'Electronics'}"
        self.assertEqual(product.import_and_validate(data), expected)

    def test_partial_data_with_partial_true(self):
        product = Product()
        data = '{"name": "Camera"}'
        result = product.import_and_validate(data)
        self.assertIn("price", result)

if __name__ == "__main__":
    unittest.main()

File: data/java/tests/test_Schematics_model_import_data.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_Schematics_model_import_data {

    @Test
    public void test_valid_data() {
        function_Schematics_model_import_data product = new function_Schematics_model_import_data();
        String data = "{\"name\": \"Laptop\", \"price\": 1000}";
        String expected = "{'name': 'Laptop', 'price': 1000, 'category': 'General'}";
        assertEquals(expected, product.import_and_validate(data));
    }

    @Test
    public void test_missing_required_field() {
        function_Schematics_model_import_data product = new function_Schematics_model_import_data();
        String data = "{\"price\": 1000}";
        String result = product.import_and_validate(data);
        // 检查错误消息中包含 "name"
        assertTrue(result.contains("name"));
    }

    @Test
    public void test_default_category() {
        function_Schematics_model_import_data product = new function_Schematics_model_import_data();
        String data = "{\"name\": \"Tablet\", \"price\": 500}";
        String nativeResult = product.import_and_validate(data);
        // 检查输出字符串中包含默认 category 值
        assertTrue(nativeResult.contains("'category': 'General'"));
    }

    @Test
    public void test_all_fields_provided() {
        function_Schematics_model_import_data product = new function_Schematics_model_import_data();
        String data = "{\"name\": \"Headphones\", \"price\": 200, \"category\": \"Electronics\"}";
        String expected = "{'name': 'Headphones', 'price': 200, 'category': 'Electronics'}";
        assertEquals(expected, product.import_and_validate(data));
    }

    @Test
    public void test_partial_data_with_partial_true() {
        function_Schematics_model_import_data product = new function_Schematics_model_import_data();
        String data = "{\"name\": \"Camera\"}";
        String result = product.import_and_validate(data);
        // 检查错误消息中包含 "price"
        assertTrue(result.contains("price"));
    }
}